### DISCUSSION TITLE
节点如何根据内容多少设置宽度

### DISCUSSION THREAD

**Discussion by sunyonghua** [antvis/X6#2422] [Q&A]
默认宽度支撑不了功能
使用方式：
节点内有名称外还有标签，每个标签都代表一个分支节点，存在一个链接桩，且分支内容不定，如何根据分支内容去动态调整节点的宽度
如图：
![image](https://user-images.githubusercontent.com/26454332/180912550-c1b209fe-2b6a-4d06-a683-8dfa417853c3.png)
有没有合适的时机，合适的方法来做这件事？

**Comment by sunyonghua** [comment]
想要的结果是可以做到
三种宽度取最大值
1.节点默认宽，2.节点名称内容宽度，3.多分支相加宽度

**Comment by sunyonghua** [comment]
![image](https://user-images.githubusercontent.com/26454332/180920477-1ad5cf70-2c8a-426c-ab40-f5fbc9f08476.png)

使用node.fit() 也没改变节点大小

```javascript 
  const initData = () => {
    graph.fromJSON(data)
    graph.getNodes().forEach((node) => {
      node.fit({ deep: true, padding: 50 })
    })
  }
```

**Comment by zmcode** [comment]
你的自定义节点不是父级吧?
新增的节点有添加到父节点中吗? addChild

**Reply by zmcode** [reply]
初始化的时候, 需要关联子父关系

**Reply by sunyonghua** [reply]
@yaojin2070 
分支不是用的x6当中的node来表示的，是react当中的node，这可能是造成node.fit()原因？
如果先不考虑分支，只考虑节点名称，貌似也没办法获取到节点名称的宽度去重新resize
现在发现一个不太靠谱的办法就是在创建节点的时候用button标签去包裹内容才能拿到真实宽度去resize

**Reply by zmcode** [reply]
![image](https://user-images.githubusercontent.com/48686959/181473381-313f6c79-659c-44b9-ae64-9e0041150f71.png)
react 和vue 都是通过svg的foreignObject来实现的, 就跟不用框架写东西一样, 只是内部的数据用框架来维护数据而已

初始化的时候, 生成的每一个react组件, 你需要先隐藏插入到dom中计算它的实际大小, 后面在统一删除掉, 就用一个promise的事情, 就是在没生成这个节点的时候, 你就需要先计算实际dom的大小, 因为需要设置这个节点的大小的, 并且这个节点需要放到父级当中, 就是parentNode.addChild(xxx), 后续新增的节点也是需要这个操作, 配合fit就可以实现需求了
